src/main/java/ch/ivyteam/ivy/maven/engine/download/URLEngineDownloader.java

```diff
@@ -80,6 +97,18 @@ private File evaluateTargetFile(URL engineUrl) {
     return downloadZip;
   }
 
+  private void wagonDownload(Repository repo, String resource, Path target) throws MojoExecutionException {
+    HttpWagon wagon = new HttpWagon();
+    try {
+      wagon.connect(repo, null, proxies);
+      wagon.get(resource, target.toFile());
+    } catch (Exception ex) {
+      throw new MojoExecutionException("Download failed from repo " + repo + " with resource " + resource, ex);
+    } finally {
+      wagon.closeConnection();
```

Comment:
null check? or is new HttpWagon() super safe?

Reply:
can't think of a situation where this is null. Either the construction fails with an exception or it's not null.